File: java/dip/AddingBorder.java
package dip;

import org.opencv.core.Core;
import org.opencv.core.Mat;
import org.opencv.imgcodecs.Imgcodecs;
import origami.Origami;

import java.io.IOException;

public class AddingBorder {
    public static void main(String[] args) throws IOException {
        Origami.init();
        Mat source = Imgcodecs.imread("data/dip/digital_image_processing.jpg");
        Mat destination = new Mat(source.rows(), source.cols(), source.type());

        int top, bottom, left, right;
        top = (int) (0.05 * source.rows());
        bottom = (int) (0.05 * source.rows());
        left = (int) (0.05 * source.cols());
        right = (int) (0.05 * source.cols());

        Core.copyMakeBorder(source, destination, top, bottom, left, right, Core.BORDER_WRAP);
        Imgcodecs.imwrite("borderWrap.jpg", destination);

        Core.copyMakeBorder(source, destination, top, bottom, left, right, Core.BORDER_REFLECT);
        Imgcodecs.imwrite("borderReflect.jpg", destination);

        Core.copyMakeBorder(source, destination, top, bottom, left, right, Core.BORDER_REPLICATE);
        Imgcodecs.imwrite("borderReplicate.jpg", destination);

    }
}

File: kotlin/sample/AddingBorder.kt
package sample


import org.opencv.core.Core.*
import org.opencv.core.Mat
import org.opencv.imgcodecs.Imgcodecs.*
import org.scijava.nativelib.NativeLoader.*
import origami.Origami

import java.io.IOException

object AddingBorder {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        Origami.init()

        val source = imread("data/dip/digital_image_processing.jpg")
        val destination = Mat(source.rows(), source.cols(), source.type())

        val top = source.rows() / 20
        val bottom = source.rows() / 20
        val left = source.cols() / 20
        val right = source.cols() / 20

        copyMakeBorder(source, destination, top, bottom, left, right, BORDER_WRAP)
        imwrite("borderWrap.jpg", destination)

        copyMakeBorder(source, destination, top, bottom, left, right, BORDER_REFLECT)
        imwrite("borderReflect.jpg", destination)

        copyMakeBorder(source, destination, top, bottom, left, right, BORDER_REPLICATE)
        imwrite("borderReplicate.jpg", destination)

    }
}
